connaisseur/config.py

```diff
@@ -27,55 +31,72 @@ def __init__(self):
+                with safe_path_func(
+                    open, self.__EXTERNAL_PATH, auth_path, "r"
+                ) as auth_file:
+                    auth_dict = {"auth": yaml.safe_load(auth_file)}
+                validator.update(auth_dict)
+        return config
 
     def __validate(self, config: dict):
-
         validate_schema(
             config,
             self.__SCHEMA_PATH,
             "Connaisseur configuration",
             InvalidConfigurationFormatError,
         )
-        notary_names = [notary.get("name") for notary in config]
-        if collections.Counter(notary_names)["default"] > 1:
-            msg = "Too many default notary configurations."
+        validator_names = [validator.get("name") for validator in config]
+        if collections.Counter(validator_names)["default"] > 1:
+            msg = "Too many default validator configurations."
             raise InvalidConfigurationFormatError(message=msg)
 
-        for notary in config:
-            key_names = [key.get("name") for key in notary.get("pub_root_keys")]
-            if collections.Counter(key_names)["default"] > 1:
-                msg = "Too many default keys in notary configuration {notary_name}."
-                raise InvalidConfigurationFormatError(
-                    message=msg, notary_name=notary.get("name")
-                )
-
-        self.notaries = [Notary(**notary) for notary in config]
+        # for notary in config:
+        #     key_names = [key.get("name") for key in notary.get("pub_root_keys")]
+        #     if collections.Counter(key_names)["default"] > 1:
+        #         msg = "Too many default keys in notary configuration {notary_name}."
+        #         raise InvalidConfigurationFormatError(
+        #             message=msg, notary_name=notary.get("name")
+        #         )
 
-    def get_notary(self, notary_name: str = None):
+    def get_validator(self, validator_name: str = None):
         """
-        Returns the notary configuration with the given `notary_name`. If `notary_name`
-        is None, the element with `name=default` is taken, or the only existing element.
+        Returns the validator configuration with the given `validator_name`. If
+        `validator_name` is None, the element with `name=default` is taken, or the only
+        existing element.
 
         Raises `NotFoundException` if no matching or default element can be found.
         """
         try:
-            if len(self.notaries) < 2:
-                return next(iter(self.notaries))
+            if len(self.validators) < 2:
+                return next(iter(self.validators))
 
-            notary_name = notary_name or "default"
+            validator_name = validator_name or "default"
             return next(
-                notary for notary in self.notaries if notary.name == notary_name
+                validator
+                for validator in self.validators
+                if validator.name == validator_name
```

Comment:
It seems to be that validator name uniqueness is not enforced, right? If so, this would return the first one with the given name. I think we should require uniqueness at validation time.

Other approach:
```python
return list(filter(lambda v: v.name == (validator_name or "default"), self.validators))[0]
```